docs/parameters/validation.md

```diff
@@ -0,0 +1,136 @@
+---
+title: Validation
+description: Validation of pipeline parameters at run time.
+---
+
+# Validation of pipeline parameters
+
+## `validateParameters()`
+
+This function takes all pipeline parameters and checks that they adhere to the specifications defined in the JSON Schema.
+
+- It does not return anything, but logs errors or warnings indicating the parameters that failed to the command line.
+- If any parameter validation has failed, it throws a `SchemaValidationException` exception to stop the pipeline.
+- If any parameters in the schema reference a sample sheet schema with `schema`, that file is loaded and validated.
+
+The function takes a single argument: the filename of a JSON Schema file.
+File paths should be relative to the root of the pipeline directory.
+Default: `nextflow_schema.json`.
+
+!!! tip
+
+    As much of the Nextflow ecosystem assumes the `nextflow_schema.json` filename, it's recommended to stick with the default, if possible.
+
+See the [Schema specification](../nextflow_schema/nextflow_schema_specification.md) for information about what validation data you can encode within the schema for each parameter.
+
+### Example
+
+The example below has a deliberate typo in `params.input` (`.txt` instead of `.csv`).
+The validation function catches this for two reasons:
+
+- The filename doesn't match the expected pattern (here checking file extensions)
+- The supplied file doesn't exist
+
+The function causes Nextflow to exit immediately with an error.
+
+=== ":material-close-thick: Output"
+
+    ```
+    --8<-- "examples/validateParameters/log.txt"
+    ```
+
+=== "main.nf"
+
+    ```groovy
+    --8<-- "examples/validateParameters/pipeline/main.nf"
+    ```
+
+=== "nextflow.config"
+
+    ```groovy hl_lines="6"
+    --8<-- "examples/validateParameters/pipeline/nextflow.config"
+    ```
+
+=== "nextflow_schema.json"
+
+    ```json linenums="1" hl_lines="17 20"
+    --8<-- "examples/validateParameters/pipeline/nextflow_schema.json"
+    ```
+
+## Unrecognized parameters
```

Comment:
missing section with `nextflow run my_pipeline --validationWarnUnrecognisedParams` ? as im seeing only `validationFailUnrecognisedParams`

Reply:
Warning unrecognised params is the default